rclcpp/src/rclcpp/duration.cpp

```diff
@@ -183,7 +183,8 @@ bounds_check_duration_scale(int64_t dns, double scale, uint64_t max)
   auto abs_dns = static_cast<uint64_t>(std::abs(dns));
   auto abs_scale = std::abs(scale);
   if (abs_scale > 1.0 &&
-    abs_dns > static_cast<uint64_t>(static_cast<long double>(max) / static_cast<long double>(abs_scale)))
+    abs_dns >
```

Comment:
How about moving `abs_dns >` to the line above?

Reply:
> How about moving `abs_dns >` to the line above?

Sure, I don't have a preference other than silencing the linter failure ;-)

Reply:
Great, please update this PR accordingly.

Reply:
> Great, please update this PR accordingly.

Done!